lib/ansible/plugins/inventory/docker_machine.py

```diff
@@ -0,0 +1,271 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2019, Ximon Eighteen <ximon.eighteen@gmail.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = '''
+    name: docker_machine
+    plugin_type: inventory
+    short_description: Docker Machine inventory source
+    requirements:
+        - L(Docker Machine,https://docs.docker.com/machine/)
+    extends_documentation_fragment:
+        - inventory_cache
+        - constructed
+    description:
+        - Get inventory hosts from Docker Machine.
+        - Uses a YAML configuration file that ends with docker_machine.(yml|yaml).
+        - The plugin sets standard host variables C(ansible_host), C(ansible_port), C(ansible_user) and C(ansible_ssh_private_key).
+        - The plugin stores the Docker Machine 'env' output variables in I(dm_) prefixed host variables.
+
+    options:
+        plugin:
+            description: token that ensures this is a source file for the C(docker_machine) plugin.
+            required: yes
+            choices: ['docker_machine']
+        daemon_required:
+            description: when true, hosts for which Docker Machine cannot output Docker daemon connection environment variables will be skipped.
+            type: bool
+            default: yes
+        running_required:
+            description: when true, hosts which Docker Machine indicates are in a state other than C(running) will be skipped.
+            type: bool
+            default: yes
+        verbose_output:
+            description: when true, include all available nodes metadata (e.g. Image, Region, Size) as a JSON object.
+            type: bool
+            default: yes
+        split_tags:
+            description: for keyed_groups add two variables as if the tag were actually a key value pair separated by a colon, instead of just a single value.
+            type: bool
+            default: no
+        split_separator:
```

Comment:
On IRC, @bcoca mentioned that this should be doable with `construct`. It is currently not possible, though, but it is better if this can be solved in a plugin-independent way in the future, than now having plugins do it differently.

@ximon18 do you mind removing this? Or is this a feature you absolutely need? With `construct`, you can do essentially the same, but not for arbitrary key-value pairs, but only for keys you know about in the inventory config. If that's ok, removing this probably speeds up getting this merged :)

Reply:
Ah, sorry. I mean `split_tags` and `split_separator`.

Reply:
I'll remove `split_tags` and `split_separator` now.